Fix streams terminology (#214)
mengelbart/rtp-over-quic-draft: draft-ietf-avtcore-rtp-over-quic.md

Senders can also transmit RTP packets in QUIC DATAGRAMs, using
a QUIC extension described in {{!RFC9221}}.
DATAGRAMs can only be used if the use of the DATAGRAM extension was successfully negotiated during the QUIC handshake.

> If the QUIC extension was negotiated using a signaling protocol, but that extension was not negotiated during the resulting QUIC handshake, an endpoint MAY close the connection with the ROQ\_EXPECTATION\_UNMET error code.

Reviewer: Why is this changed to a normative "MAY"? But if we are thinking about normative behavior, why would an endpoint not close the connection immediately? It seems odd to notice a mismatch at signaling time, but not give the peer endpoint any indication that all is not right, until the peer endpoint tries to use the QUIC extension. My understanding of [Section 12.4 in RFC 9000](https://datatracker.ietf.org/doc/html/rfc9000#name-frames-and-frame-types) is that the result would be a connection error of type FRAME_ENCODING_ERROR, and that doesn't point directly to a signaling/negotiation mismatch, which is the real problem.

Author: At the end of [section 3.2](https://mengelbart.github.io/rtp-over-quic-draft/draft-ietf-avtcore-rtp-over-quic.html#section-3.2), we say that an application can use streams instead. FRAME_ENCODING_ERROR would only happen if the QUIC stack is sending datagrams even though the datagram extension was not negotiated at connection establishment. In that case, I would expect the QUIC stack to return an error when the application is trying to send datagrams.

Reviewer: > At the end of [section 3.2](https://mengelbart.github.io/rtp-over-quic-draft/draft-ietf-avtcore-rtp-over-quic.html#section-3.2), we say that an application can use streams instead. FRAME_ENCODING_ERROR would only happen if the QUIC stack is sending datagrams even though the datagram extension was not negotiated at connection establishment. In that case, I would expect the QUIC stack to return an error when the application is trying to send datagrams.

Oh, right. This is a variation of the same conversation we were having at the Hackathon in San Francisco - sorry for MY confusion. 

One thing that doesn't help me stay unconfused, is that we are talking about a specific extension here, but the text sounds more general. At the end of [Section 3.2](https://mengelbart.github.io/rtp-over-quic-draft/draft-ietf-avtcore-rtp-over-quic.html#name-rtp-with-quic-streams-quic-), we explain this more clearly than we're saying in this text. The point in Section 3.2 is "you have a decision to make. How badly do you need to use DATAGRAMs?" 

```suggestion
If the DATAGRAM extension was negotiated using a signaling protocol, but was not also negotiated during the resulting QUIC handshake, an endpoint can close the connection with the ROQ\_EXPECTATION\_UNMET error code.
```

QUIC DATAGRAMs are ACK-eliciting packets, which means that an acknowledgment is
triggered when a DATAGRAM frame is received. Thus, a sender can assume that an
RTP packet arrived at the receiver or was lost in transit, using the QUIC
acknowledgments of QUIC DATAGRAM frames. In the following, an RTP packet is
regarded as acknowledged when the QUIC DATAGRAM frame that carried the RTP
packet was acknowledged.

## RoQ Streams {#roc-s}

For RTP packets that are sent over QUIC streams, an RTP packet is considered

> acknowledged after all frames that carried parts of the RTP packet were

Reviewer: ```suggestion
acknowledged after all STREAM frames that carried parts of the RTP packet were
```
